flytekit/deck/deck.py

```diff
@@ -145,14 +145,17 @@ def _get_deck(
 
 def _output_deck(task_name: str, new_user_params: ExecutionParameters):
     ctx = FlyteContext.current_context()
-    if ctx.execution_state.mode == ExecutionState.Mode.TASK_EXECUTION:
-        output_dir = ctx.execution_state.engine_dir
-    else:
-        output_dir = ctx.file_access.get_random_local_directory()
-    deck_path = os.path.join(output_dir, DECK_FILE_NAME)
-    with open(deck_path, "w") as f:
+    local_dir = ctx.file_access.get_random_local_directory()
+    local_path = os.path.join(local_dir, DECK_FILE_NAME)
+    with open(local_path, "w") as f:
         f.write(_get_deck(new_user_params, ignore_jupyter=True))
-    logger.info(f"{task_name} task creates flyte deck html to file://{deck_path}")
+    logger.info(f"{task_name} task creates flyte deck html to file://{local_path}")
+    if ctx.execution_state.mode == ExecutionState.Mode.TASK_EXECUTION:
+        remote_path = os.path.join(new_user_params.output_prefix, DECK_FILE_NAME)
+        kwargs: typing.Dict[str, str] = {
+            "ContentType": "text/html",
+        }
+        ctx.file_access.put_data(local_path, remote_path, **kwargs)
```

Comment:
by doing this we're going to copy the deck file twice, no?

Reply:
no, we create a deck file at line 151, and then upload it to s3.